controller/src/main/java/io/pravega/controller/server/rest/impl/RESTServerConfigImpl.java

```diff
@@ -21,13 +21,19 @@
 public class RESTServerConfigImpl implements RESTServerConfig {
     private final String host;
     private final int port;
+    private final boolean tlsEnabled;
+    private final String keyFilePath;
+    private final String keyFilePasswordPath;
 
     @Builder
-    RESTServerConfigImpl(final String host, final int port) {
+    RESTServerConfigImpl(final String host, final int port, boolean tlsEnabled, String keyFilePath, String keyFilePasswordPath) {
```

Comment:
Should keyFilePath or keyPasswordPath be non-null, non-empty? Consider validating them.

Reply:
They need not be non-null, non-empty. In case of `tlsEnabled` false, they are not used at all.

Reply:
I think @andreipaduroiu is requesting that you add more pre-conditions here.

Reply:
@fpj, thanks for clarifying. Added more preconditions.